lib/go/test/service_test.go

```diff
@@ -401,6 +401,76 @@ func TestContracts(t *testing.T) {
 
 	})
 
+	t.Run("Should check if payer has sufficient balance to execute tx", func(t *testing.T) {
```

Comment:
@joshuahannan  
I cannot test a new functionality because I'm running into this [check](https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowFees.cdc#L111) all the time. I noticed that this [function](https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowFees.cdc#L109) always returns zero. [This statement](https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowStorageFees.cdc#L175) is zero because [this one](https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowStorageFees.cdc#L133) is always true. 

That's why I submitted a tx `GenerateChangeStorageFeeParametersScript()` in this test. I wanted to set `FlowStorageFees.storageMegaBytesPerReservedFLOW` but it looks like I'm missing sth as I'm still falling through [this check](https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowFees.cdc#L111).

Can you help me with it?

Reply:
I want to be able to adjust script arguments and amount of flow on the account to test that the script works as expected. For now I cannot to this because of the above-described problem.

Reply:
Are you saying that the transaction you are sending to set `storageMegaBytesPerReservedFLOW` is failing because of that check? Even if that check returns, the transaction should still be able to execute because it returns `true` for `canExecuteTransaction`. So you should be able to test after setting that field.

Also, I believe there is a way to start the emulator with transaction and storage fees turned on, so you might be able to just do that instead. Let me see if I can find that

Reply:
As you can see [here](https://github.com/onflow/flow-core-contracts/blob/master/lib/go/test/test.go#L85), we normally start the emulator with storage fees disabled, but you can probably set that to true and then it will be enabled and you won't have to set anything yourself.

There is probably an option for transaction fees too

Reply:
You can find all your emulator options here: https://github.com/onflow/flow-emulator/blob/master/emulator/blockchain.go#L118

Reply:
> Are you saying that the transaction you are sending to set `storageMegaBytesPerReservedFLOW` is failing because of that check? Even if that check returns, the transaction should still be able to execute because it returns `true` for `canExecuteTransaction`. So you should be able to test after setting that field.
> 
> Also, I believe there is a way to start the emulator with transaction and storage fees turned on, so you might be able to just do that instead. Let me see if I can find that

I'm saying that `canExecuteTransaction` value is always `true` no matter what execution effort, gas limit or payer balance is

Reply:
When I set `StorageLimitEnabled = true` I can see that the values of `VerifyPayerBalanceResult` start changing and I'm no longer running into that check, which is good. 
However, the `canExecuteTransaction` is still always equal to `true` as `requiredBalance` is always bigger or equal to the actual payer's balance. Even if I don't mint any tokens, account's balance is "enough to pay for tx". I tried to set both `inclusionEffort` and `gasLimit` to different values and it doesn't change the behavior. Is it expected?

Reply:
You are probably hitting this check: 
```
if minimumRequiredBalance == UFix64(0) {
            // If the required balance is zero exit early.
```
https://github.com/onflow/flow-core-contracts/blob/master/contracts/FlowFees.cdc#L111

In the FVM you need both  context options 
```
fvm.WithTransactionFeesEnabled(true),
fvm.WithAccountStorageLimit(true),
```
and bootstrap options
```
fvm.WithTransactionFee(fvm.DefaultTransactionFees),
fvm.WithStorageMBPerFLOW(fvm.DefaultStorageMBPerFLOW),
fvm.WithMinimumStorageReservation(fvm.DefaultMinimumStorageReservation),
fvm.WithAccountCreationFee(fvm.DefaultAccountCreationFee),
```

Not exactly sure how to set that up in a test in the core-contracts but I believe you have to set the emulator options in which case  you can use the corresponding emulator options:
```
WithStorageLimitEnabled
WithMinimumStorageReservation // this covers both WithMinimumStorageReservation and WithAccountCreationFee
WithStorageMBPerFLOW
WithTransactionFeesEnabled
```

Reply:
Hey @janezpodhostnik. I did set up the options you recommended though the payer has still enough balance to pay for tx. I tried to execute some simple transaction and set my custom-created account as a tx payer but I'm running into errors saying it doesn't have sufficient signatures. Can u please take a look at it? I pushed a commit with the latest changes.

Reply:
Just to be clear: I want to test the case where the payer has sufficient balance and insufficient one to pay for tx.
The issue is that the balance is always sufficient even if I don't mint any tokens and set inclusion effort to a big value.
I decided to execute some simple transaction (as a payer) to reduce the amount of Flow on the account so that I have insufficient balance and can test this case.

For now, I have a problem with signatures. However, I'd like to hear any advice from you If the thing I'm making does make sense.

Reply:
@joshuahannan can you please take a look into this while Janez is off?